service/reducers/entities/files.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {FilesTypes, UsersTypes} from 'service/constants';
+
+function files(state = {}, action) {
+    switch (action.type) {
+    case FilesTypes.RECEIVED_FILES_FOR_POST: {
+        const filesById = action.data.reduce((filesMap, file) => {
+            return {...filesMap,
+                [file.id]: file
+            };
+        }, {});
+        return {...state,
+            ...filesById
+        };
+    }
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+    default:
+        return state;
+    }
+}
+
+function fileIdsByPostId(state = {}, action) {
```

Comment:
@hmhealey Legacy posts, eh? So there are posts that have file attachments but no `file_ids`? Do they have the `filenames` field instead?
Is that the reason for this code?
```
if ((post.file_ids && post.file_ids.length > 0) || (post.filenames && post.filenames.length > 0)) {
```
I saw that in the webapp but didn't think it was necessary to copy over. Should I also make this same check for legacy posts?